go/vt/sidecardb/sidecardb.go

```diff
@@ -396,7 +396,7 @@ func (si *schemaInit) getCurrentSchema(tableName string) (string, error) {
 // or an ALTER if the table exists but has a different schema.
 func (si *schemaInit) findTableSchemaDiff(tableName, current, desired string) (string, error) {
 	hints := &schemadiff.DiffHints{
-		TableCharsetCollateStrategy: schemadiff.TableCharsetCollateIgnoreAlways,
+		TableCharsetCollateStrategy: schemadiff.TableCharsetCollateIgnoreEmpty,
```

Comment:
LGTM and good catch. We might also just as well "upgrade" to `TableCharsetCollateStrict` - but as this works then that's fine.